src/vs/editor/common/model/tokenizationTextModelPart.ts

```diff
@@ -69,12 +69,17 @@ export class TokenizationTextModelPart extends TextModelPart implements ITokeniz
 		);
 	}
 
+	public override dispose(): void {
+		this._languageRegistryListener.dispose();
+		this._tokenization.dispose();
+		super.dispose();
+	}
+
```

Comment:
It would be cleaner to call `this._register` in the ctor when we instantiate these readonly fields.